### ISSUE TITLE
Incorrect Finger Movements in Real-World Deployment

### ISSUE THREAD

**Issue by Hao-yang-Wang** [leap-hand/LEAP_Hand_Sim#30]
Hi. When I running the real-world deployment program, the thumb initially seems to move to an incorrect angle. After forward propagation begins, the remaining four fingers start moving towards the back of the hand. However, the ros_example.py program is working fine, with all joints zeroed and the palm in an extended position.

This is the initial position look like: 
![leaphand_init](https://github.com/user-attachments/assets/c378fa68-a690-4b61-87c6-f1f20d2dbdf8)

This is the running position look like: 
![leaphand_run](https://github.com/user-attachments/assets/7b9ded88-d09c-49bb-abed-56d70c454cf6)

Could you confirm if the motor IDs corresponding to the joints in the program match those from the calibration in the video tutorial?

Thank you for your program and opensource hand, and I look forward to your response.

**Comment by Lucky7Chess** [comment]
Your thumb does not look like its mounted correctly, it should not be spun out that far in the initial pose.  Also, the RL policy will not work without a cube in it because it is actively sensing the position of the cube to deploy the policy correctly.  The IDs definitely match what the simulation/video are if you IDed them correctly.

**Comment by Hao-yang-Wang** [comment]
Hi Kenny, thank you for your quick response. I've fixed the issue with the thumb and printed a cube with a side length of 7.5 cm. Everything seemed perfect when I started testing, but after running for about 1 second, the three fingers gradually moved toward the back of the hand and eventually got stuck. Do you have any suggestions for this situation?

https://github.com/user-attachments/assets/54f54a55-76f2-4c2e-bcb3-3c43b0410627

**Comment by Lucky7Chess** [comment]
Try tuning the PID values in real, not sure why it's not feeling the cube at all there.

**Comment by lixiaolong04** [comment]
> 嗨，Kenny，感谢您的快速回复。我已经修复了 thumb 的问题，并打印了一个边长为 7.5 厘米的立方体。当我开始测试时，一切似乎都很完美，但是在运行了大约 1 秒钟后，三根手指逐渐向手背移动，最终卡住了。对于这种情况，您有什么建议吗？
> 
>  leaphand.mp4

Hello, have you solved it? You can try to lower the current of each steering gear

**Comment by Hao-yang-Wang** [comment]
> Hello, have you solved it? You can try to lower the current of each steering gear.

Unfortunatly, I didn't solved it yet, these days I working on another project. I'd like to try it, thank you for your advise!

**Comment by Hao-yang-Wang** [comment]
> Hello, have you solved it? You can try to lower the current of each steering gear

I think I found the solution. I modified line 41 in deploy.py
`self.action_scale = 1 / 24` to `self.action_scale = 1 / 18`
Everything looks perfect afterwards.

https://github.com/user-attachments/assets/a8603335-0439-4635-85ca-f8e27dea8c90

I also tried adjusting the PID parameters and current limits, but the effect was not significant. As the author mentioned, if the fingers cannot firmly grasp the block, it won't work because the pose of the block cannot be detected. Through manually adjusting the contact between the block and the fingers, I notice that middle finger must make firm contact with the block when moving toward the palm; otherwise, the mentioned issue may occur. It seems like adjusting the action scale can reliably solve this issue, I guess adjustments may be needed due to motor model differences. 

Additionally, ensuring proper friction and maintaining a horizontal palm angle are also key to success.

Once again, thank you to the author for open-sourcing this excellent project.

**Comment by lixiaolong04** [comment]
> > Hello, have you solved it? You can try to lower the current of each steering gear
> 
> I think I found the solution. I modified line 41 in deploy.py to Everything looks perfect afterwards.`self.action_scale = 1 / 24``self.action_scale = 1 / 18`
> 
>  leaphand.mp4 
> I also tried adjusting the PID parameters and current limits, but the effect was not significant. As the author mentioned, if the fingers cannot firmly grasp the block, it won't work because the pose of the block cannot be detected. Through manually adjusting the contact between the block and the fingers, I notice that middle finger must make firm contact with the block when moving toward the palm; otherwise, the mentioned issue may occur. It seems like adjusting the action scale can reliably solve this issue, I guess adjustments may be needed due to motor model differences.
> 
> Additionally, ensuring proper friction and maintaining a horizontal palm angle are also key to success.
> 
> Once again, thank you to the author for open-sourcing this excellent project.

Yes, your hand must be in contact with the object; otherwise, it won't move properly. Wish you all the best for your future experiments!

**Comment by lycopene123** [comment]
I had the same issue with a 3D-printed cube. The surface was too smooth for stable contact. I applied some anti-slip tape, and it works much better now.

**Comment by asrasarwat** [comment]
Hi 👋

I’m a PhD student working on robotic hand control and I’m currently using the LEAP Hand. I’m particularly interested in grasping small objects.

I wanted to ask if you have any code, examples, or guidance related to small-object grasping with the LEAP Hand? Especially anything related to precision grasping or control strategies would be really helpful.

Thank you in advance for your support! 🙏

**Comment by asrasarwat** [comment]
I did chang deploy self.action_scale = 1 / 24``self.action_scale = 1 / 18 but still my hand after grasping move away.
